projects/step-core/src/lib/components/expression-input/expression-input.component.ts

```diff
@@ -0,0 +1,63 @@
+import { Component, EventEmitter, forwardRef, Input, Output, ViewEncapsulation } from '@angular/core';
+import { ControlValueAccessor, NG_VALUE_ACCESSOR } from '@angular/forms';
+import { DialogsService } from '../../shared';
+
+type OnChange = (expression: string) => void;
+type OnTouch = () => void;
+
+@Component({
+  selector: 'step-expression-input',
+  templateUrl: './expression-input.component.html',
+  styleUrls: ['./expression-input.component.scss'],
+  encapsulation: ViewEncapsulation.None,
+  providers: [
+    {
+      provide: NG_VALUE_ACCESSOR,
+      useExisting: forwardRef(() => ExpressionInputComponent),
+      multi: true,
+    },
+  ],
+})
+export class ExpressionInputComponent implements ControlValueAccessor {
+  @Input() label?: string;
+  @Input() tooltip?: string;
+  @Input() invalid: boolean = false;
+  @Input() required: boolean = false;
```

Comment:
Same as above. If we are using angular's forms, we should rely on validation mechanism provided by framework.
Otherwise it creates a possibility of contradictions like:
- Form control is valid and non required, but input flags are set is true
- Vice versa. Form control is required and invalid, but input flags are set is false